### PR TITLE
[MPE][Wasm] Early media error causes autoplay to fail - playground issue

### PR DESCRIPTION
**Pull request by rafael-rosa-knowcode**
GitHub Issue (If applicable): 
https://github.com/unoplatform/uno/issues/12533

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

The publish url on playground show the incorrect icon for play or pause.
See the link https://playgroundcanary.z19.web.core.windows.net/#bcb300bd
image

## What is the new behavior?

Now the icon shows the Play when the video is blocked by the browser on the autoplay.

### <samp>🤖 Generated by Copilot at 801f41c</samp>

This pull request improves the media playback state management of the `HtmlMediaPlayer` class for WebAssembly by introducing a new `HtmlMediaPlayerState` enum and using it instead of the `IsPause` field. It also updates the `MediaPlayerExtension` class and removes some unused or redundant code.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/AddIns/Uno.UI.MediaPlayer.WebAssembly/HtmlMediaPlayerState.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Uno.UI.Media;
+
+public enum HtmlMediaPlayerState
```
Comment (jeromelaban) [inline src/AddIns/Uno.UI.MediaPlayer.WebAssembly/HtmlMediaPlayerState.cs:9]: Let's make this internal.

File: src/AddIns/Uno.UI.MediaPlayer.WebAssembly/HtmlMediaPlayer.cs [hunk 2]
```diff
@@ -319,9 +336,28 @@ private void OnHtmlMetadataLoaded(object sender, EventArgs e)
...
+		{
+			this.Log().Debug($"Media Suspended [{Source}]");
+		}
+
+		_activeElement = IsVideo ? _htmlVideo : IsAudio ? _htmlAudio : default;
+		_activeElementName = IsVideo ? "Video" : IsAudio ? "Audio" : "";
+		if (_activeElement != null)
+		{
+			PlayerState = NativeMethods.GetPaused(_activeElement.HtmlId) ? HtmlMediaPlayerState.Paused : HtmlMediaPlayerState.Playing;
+			_activeElement.SetCssStyle("visibility", "visible");
```
Comment (jeromelaban) [inline src/AddIns/Uno.UI.MediaPlayer.WebAssembly/HtmlMediaPlayer.cs:355]: Why do we need to change the visibility in this case?

File: src/AddIns/Uno.UI.MediaPlayer.WebAssembly/HtmlMediaPlayerState.cs (hunk 1 above)
Comment (rafael-rosa-knowcode) [reply src/AddIns/Uno.UI.MediaPlayer.WebAssembly/HtmlMediaPlayerState.cs:9]: Done.

File: src/AddIns/Uno.UI.MediaPlayer.WebAssembly/HtmlMediaPlayer.cs (hunk 2 above)
Comment (rafael-rosa-knowcode) [reply src/AddIns/Uno.UI.MediaPlayer.WebAssembly/HtmlMediaPlayer.cs:355]: I will remove. it is overzealous, to show the image.
For once in the test Suspended came before SourceLoaded.

**Pull request merged**